### PR TITLE
Electricity resistance tweaks re #10392

### PR DESCRIPTION
**Pull request by Eighth-of-Third**
#10392

Essentially, when checking if an item is conductive assume that it is being held by a part made of non-conductive material if it has a non-conductive material. Adds a CONDUCTIVE flag for the cases where this assumption is not correct. Non-conductive material is considered to be any material with an elec_resist of 2 or greater; metal has an elec_resist of 0, flesh has an elec_resist of 1, wood has an elec_resist of 2.

This means if you're hitting a shocker zombie with a barbed wire bat (Steel and Wood) you won't be shocked. However, if you've got the Electroshock Unit and hit zombies with a barbed wire bat you won't be able to shock them.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
The conductive flag isn't used and so shouldn't be added in the PR.

File: src/item.cpp [hunk 1]
```diff
@@ -3365,14 +3365,18 @@ bool item::conductive() const
...
         return false;
     }
+    
+    if (has_flag( "CONDUCTIVE" ) ) {  
+        return true;
+    }
 
-    // If any material does not resist electricity we are conductive.
+    // If any material has electricity resistance equivilant to wood we are not conductive.
     for (auto mat : made_of_types()) {
```
Comment (Coolthulhu) [inline src/item.cpp]: The loop and the returns can be rewritten as:
```
const auto mats = made_of_types();
return std::all_of( mats.begin(), mats.end(), []( const material_type *mt ) {
    return mt->elec_resist() >= 2;
} );
```

**Comment by Izicata** [comment]
Alright. I guess anyone who wants to add a conductive weapon with wood in it can re-add the flag.

**Comment by Coolthulhu** [comment]
Did you look over weapons that will be affected?

The hafted weapons like axes and spears are probably OK, but there may be edge cases like ceramic knives.

**Comment by Izicata** [comment]
As far as I can tell, the only ceramic knife in the game is the RM42 and it's got a plastic grip.

**Comment by flitvious** [comment]
Shouldn't very short weapons be conductive? If you stab a zombie with a pocket knife, it would go in up to the hilt, so your hand touches the zombie.

I've also thought that shockers create those arcs of electricity. So if you hit a shocker with a katana, electricity goes through the blade, then arcs to your hand through the small air gap created by the guard.

**Comment by thebombsauce** [comment]
I'm pretty inexperienced in physics but its my understanding that current can basically run through anything if the voltage is high enough and that the distinction between conductors and insulators is a little blurry due to this. So I think it would be fine to just set a baseline and add a little bit of strategy to the game as players decide whether a weapon is appropriate for fighting shockers.

**Comment by Izicata** [comment]
>Shouldn't very short weapons be conductive? If you stab a zombie with a pocket knife, it would go in up to the hilt, so your hand touches the zombie.

Very short weapons without a good guard, stuff like pocket knives and screwdrivers. Yeah, that makes sense.

>I've also thought that shockers create those arcs of electricity. So if you hit a shocker with a katana, electricity goes through the blade, then arcs to your hand through the small air gap created by the guard.

Katanas generally do have a solid wooden spacer between the blade and the hilt; some don't, but most do.

**Comment by Coolthulhu** [comment]
> Katanas generally do have a solid wooden spacer between the blade and the hilt; some don't, but most do. 

If katanas are to be non-conductive, the primary reason for zapback defense - making shockers require worse weapons - is rendered useless.

Also, all diamond weapons need to retain their pre-diamondization conductivity, since only the cutting part of the edge is covered in diamond.

**Comment by Coolthulhu** [comment]
It would probably be easier to just add a "NONCONDUCTIVE" flag and invert the check, so that weapons with metal elements are conductive by default, then add the flag to spears and axes.

**Comment by Izicata** [comment]
>If katanas are to be non-conductive, the primary reason for zapback defense - making shockers require worse weapons - is rendered useless.

Actually, none of the sword-type weapons have the wood material. So this PR won't do that.

**Comment by Izicata** [comment]
>It would probably be easier to just add a "NONCONDUCTIVE" flag and invert the check, so that weapons with metal elements are conductive by default, then add the flag to spears and axes.

We'd still need a CONDUCTIVE flag for stuff like the glass shiv and sharpened toothbrush. They don't have guards, but they're made of fully nonconductive materials.

**Comment by Coolthulhu** [comment]
Errors on load

**Comment by Izicata** [comment]
As #17909 was merged, this is a duplicate.

**Pull request closed without merging**